core/src/main/java/org/elasticsearch/index/translog/Translog.java

```diff
@@ -434,6 +439,7 @@ public Location add(Operation operation) throws IOException {
                 ensureOpen();
                 Location location = current.add(bytes);
                 assert assertBytesAtLocation(location, bytes);
+                lastWriteLocation = location;
```

Comment:
just make the size 0 since the offset will be lastWriteOffset+size in the current writer?